File: AI.Core/OperatorsImplimentation/BasicOperators/If.cs
namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    class If : Operator
    {
        public If(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 3;
        }

        public override void Action()
        {
            var currentValue = EnterContacts[0].Value;
            var downLimit = EnterContacts[1].Value;
            var upperLimit = EnterContacts[2].Value;
            if (currentValue >= downLimit && currentValue < upperLimit)
            {
                ExitContacts.SetValue(1);
            }
            else
            {
                ExitContacts.SetValue(0);
            }
        }
    }
}

File: NativeCore/If.cpp
#include "If.h"
#include "stdafx.h"

namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				If::If(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(3);
				}

				void If::Action()
				{
					auto currentValue = EnterContacts[0]->Value;
					auto downLimit = EnterContacts[1]->Value;
					auto upperLimit = EnterContacts[2]->Value;
					if (currentValue >= downLimit && currentValue < upperLimit)
					{
						ExitContacts->SetValue(1);
					}
					else
					{
						ExitContacts->SetValue(0);
					}
				}
			}
		}
	}
}
